provider/cluster/kube/client.go

```diff
@@ -786,3 +787,7 @@ func (c *client) deploymentsForLease(ctx context.Context, lid mtypes.LeaseID) (m
 
 	return serviceStatus, nil
 }
+
+func (c *client) KubeVersion() (*version.Info, error) {
+	return c.kc.(*kubernetes.Clientset).ServerVersion()
```

Comment:
It seems like this method exists on an interface on its own. It's more robust to just cast to that interface type. Also we should check for the cast failing and just return an error.

https://github.com/kubernetes/client-go/blob/master/discovery/discovery_client.go#L120

Reply:
Yes, always use methods where available.

Take the time to do things right @arijitAD

Reply:
Fixed